spell/spell-reviewer-mainnet-checklist.md

```diff
@@ -1,369 +1,372 @@
     * [ ] External calls for SubDAO content are NOT delegate call
     * [ ] Code does not have untoward behavior within the scope of Maker Core Contracts (e.g. up to the SubDAO proxy)
-* [ ] External Contracts Calls (Not SubDAOs, e.g. Starknet)
-  * [ ]  Target Contract don't block spell execution
-  * [ ]  External call is NOT delegate call
-  * [ ]  Target Contract doesn't have permissions on the Vat
-  * [ ]  Target Contract doesn't do anything untoward (e.g. interacting with unsafe contracts)
-  * [ ]  Contracts deployed via `CREATE2` (e.g. if it looks like a vanity address) do not have `selfdestruct` in their code
-  * [ ]  MCD Pause Proxy doesn't give any approvals
-  * [ ]  All possible actions of the Target Contract are documented
-  * [ ]  Target contract is not upgradable
-  * [ ]  Target Contract is included in the ChainLog
-  * [ ]  Test Coverage is comprehensive
-* [ ] ChainLog
-  * [ ] Increment ChainLog version based on update type
-    * [ ] Major -> New Vat (++.0.0)
-    * [ ] Minor -> Core Module (DSS) Update (e.g. Flapper) (0.++.0)
-    * [ ] Patch -> Collateral addition or addition/modification (0.0.++)
-* [ ] `addresses_mainnet.sol` matches spell code
+* IF external contracts calls present (Not SubDAOs, e.g. Starknet)
+  * [ ] Target Contract don't block spell execution
+  * [ ] External call is NOT delegate call
+  * [ ] Target Contract doesn't have permissions on the Vat
+  * [ ] Target Contract doesn't do anything untoward (e.g. interacting with unsafe contracts)
+  * [ ] Contracts deployed via `CREATE2` (e.g. if it looks like a vanity address) do not have `selfdestruct` in their code
+  * [ ] MCD Pause Proxy doesn't give any approvals
+  * [ ] All possible actions of the Target Contract are documented
+  * [ ] Target contract is not upgradable
+  * [ ] Target Contract is included in the ChainLog
+  * [ ] Test Coverage is comprehensive
+* IF spell interacts with ChainLog
+  * [ ] ChainLog version is incremented based on update type
+    * Major -> New Vat (++.0.0)
+    * Minor -> Core Module (DSS) Update (e.g. Flapper) (0.++.0)
+    * Patch -> Collateral addition or addition/modification (0.0.++)
+  * [ ] New addresses are added to the `addresses_mainnet.sol`
+  * [ ] Changes are tested via `testNewOrUpdatedChainlogValues`
 * [ ] Ensure every spell variable is declared as `public`/`internal`
-* [ ] Ensure `immutable` visibility is only used when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress` and `constant` is used instead for static addresses
+* [ ] Ensure `immutable` visibility is only used when fetching addresses from the `ChainLog` via `DssExecLib.getChangelogAddress(key)` and `constant` is used instead for static addresses
   * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) unless archive patterns permit otherwise (Such as `MKR`)
```

Comment:
```suggestion
  * [ ] Fetch addresses as type `address` and wrap with `Like` suffix interfaces inline (when making calls) UNLESS archive patterns permit otherwise (Such as `MKR`)
```